### PR TITLE
add PropertyDefinitionList evaluation order test

### PR DESCRIPTION
**Pull request by michaelficarra**
This test is being added because the committee is considering changing this evaluation order (as discussed at the May 2016 Munich meeting). The consequences of this spec change will be more clear as a test change than a simple test addition.

/cc @leobalter

### CODE REVIEW

File: test/language/expressions/object/computed-property-evaluation-order.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/*---
```
Comment (leobalter) [inline test/language/expressions/object/computed-property-evaluation-order.js]: This is missing the copyright section.

some equivalent comment is necessary:

```js
// Copyright (C) 2016 Michael Ficarra. All rights reserved.
// This code is governed by the BSD license found in the LICENSE file.
```

**Comment by leobalter** [comment]
missing a small detail, the tests LGTM.

**Comment by michaelficarra** [comment]
Added the copyright comment.

**Comment by leobalter** [comment]
LGTM.

cc @tcare for double sign off

File: test/language/expressions/object/computed-property-evaluation-order.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+---*/
+
+var counter = 0;
+var o = {
+  [++counter]: ++counter,
+  [++counter]: ++counter,
+  [++counter]: ++counter,
+}
+
+var keys = Object.keys(o).sort();
```
Comment (leobalter) [inline test/language/expressions/object/computed-property-evaluation-order.js]: I believe `.sort()` is not necessary here, this is a detail that won't break the tests and can be fixed during merge.

Comment (michaelficarra) [inline test/language/expressions/object/computed-property-evaluation-order.js]: I wanted it to work in an implementation that doesn't necessarily implement `Object.keys` ordering properly. This is not an `Object.keys` test.

Comment (leobalter) [inline test/language/expressions/object/computed-property-evaluation-order.js]: We have to assume Object.keys is acting according to the specs, which the order is defined.

If this built-in becomes a problem, I believe pure assertions over the expected properties will solve this in a better fashion:

```js
assert.sameValue(o[1], 2);
assert.sameValue(o[3], 4);
assert.sameValue(o[5], 6);
```

If necessary, Object.keys could still be used to compare the assigned keys, which I believe is overkill, but still a valid test:

```js
assert(compareArray(Object.keys(o), [1, 3, 5]));
```

**Comment by michaelficarra** [comment]
@tcare What needs to be done here? I'm okay with either implementation. They're both correct, and I believe the current implementation is more readable because it is more explicit. Let me know if I need to change it though.

**Comment by tcare** [comment]
I spoke with @leobalter and he'd prefer you avoid .sort() before merging, so I'll wait for his final signoff. Either of his two suggestions are fine with me.

**Comment by leobalter** [comment]
As I also spoke with @jugglinmike and we agreed there's no need to use .sort here. In terms of implementation dependent definitions, .sort is far more susceptible to a different reordering. `Object.keys` API has a pretty defined spec, any different reordering would be an error which is not a responsibility for this test. This is indeed not a test for Object.keys.

**Comment by michaelficarra** [comment]
I've removed the `.sort()` call.

**Comment by leobalter** [comment]
Thank you.

**Pull request merged**